File: LearnCSharpTests/Library/ExtensionsTests.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="ExtensionsTests.cs" company="Beatrix Klebe">
//   2018 Beatrix Klebe
// </copyright>
// <summary>
//   Defines the ExtensionsTests type.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace LearnCSharpTests.Library
{
    using System;
    using System.IO;

    using LearnFSharpLibrary;

    using Xunit;

    /// <summary>
    /// The extensions tests.
    /// </summary>
    public class ExtensionsTests
    {
        /// <summary>
        /// The print string equal.
        /// </summary>
        [Fact]
        public void PrintStringEqual()
        {
            var consoleOutput = new StringWriter();
            Console.SetOut(consoleOutput);
            const string TestString = "This is a test.";
            TestString.Print();
            Assert.Equal(TestString + Environment.NewLine, consoleOutput.ToString());
        }
    }
}

File: LearnFSharpTests/Library/ExtensionsTests.fs
module LearnFSharpTests.Library.ExtensionsTests

open System
open System.IO
open Xunit
open LearnFSharpLibrary.Extensions

[<Fact>]
let PrintTest() =
    let consoleOutput = new StringWriter()
    Console.SetOut consoleOutput
    let testString = "This is a test."
    testString.Print()
    Assert.Equal(testString + Environment.NewLine, consoleOutput.ToString())